### ISSUE TITLE
Calling click on an option element inside a select doesn't update the model, in Firefox only

### ISSUE THREAD

**Issue by juliemr** [angular/protractor#480]
Sigh.

**Comment by shaoping** [comment]
```
element(by.cssContainingText('select#my-id option', 10)).click();
```
did not work with firefox

**Comment by Jacq** [comment]
Same problem here, working with latest and previous Chrome but not with Firefox using either click or sendkeys

**Comment by chuckcfs** [comment]
has there been any progress with this? I'm having the same issue, I've work around it using the solution proposed in the forum for the webdriver bug, it works fine locally however when running with Travis CI I got an "angular could not be found in the window error" any ideas?

**Comment by gitjul** [comment]
had the same issue, worked locally but not on CircleCI

**Comment by chuckcfs** [comment]
I already find a workaround that works for me under TravisCI and locally

I do a regular:

```
element( by.css( '.select option[value="some_value"]' ) ).click();
```

And then in order for the model to update I do this little hack

```
browser.actions().sendKeys( protractor.Key.ENTER ).perform();
```

and that does it, the model get's updated and the tests work as expected, it does get executed a little slow like any other ```browser.actions()``` call but it works. 

Hope it helps somebody out there.

**Comment by iljapavlovs** [comment]
this is very annoying

**Comment by pmowrer** [comment]
Thanks @chuckcfs, that helped us!

On a side-note, this problem is intermittent for us. In a `<select>` with all US states as options, only certain state option ranges would exhibit this behavior. Perhaps it has to do with whether the given option requires the list of options to be scrolled.

**Comment by NathanWalker** [comment]
I'm using Angular2 and it appears @chuckcfs workaround does not work. I've tried dozens of different variations to get a dropdown selection to trigger change detection and nothing has worked.
Any additional thoughts?

Here are just *some* of the things I've tried ('path to select' = dom path to a select dropdown):
```
element(by.css(`path to select option[value="the-value-i-want"]`)).click();
browser.actions().sendKeys( protractor.Key.ENTER ).perform();
```

```
// where item = value i want
element(by.css('path to select')).all(by.tagName('option'))
    .then(function findMatchingOption(options: any) {
      options.some(function(option: any) {
        option.getText().then(function doesOptionMatch(text: string) {
          if (text.indexOf(item) !== -1) {
            desiredOption = option;
            return true;
          }
        });
      });
    })
    .then(function clickOption() {
      if (desiredOption) {
        desiredOption.click();
        // I've tried with and without this...
        browser.actions().sendKeys( protractor.Key.ENTER ).perform();
      }
    });
```

```
element(by.css('path to select')).$(`option[value="the-value-i-want"]`).click();
browser.actions().sendKeys( protractor.Key.ENTER ).perform();
```

All of them actually select the right option in the dropdown in the testing browser (which is **Chrome** actually) but none of them work to fire change detection as nothing changes based on what *should* change when changing the dropdown manually with my pointer/mouse.

**Comment by NathanWalker** [comment]
Ok thanks to @AlexLovato, that workaround works for me. Just a big guessing game I guess.
This works in **Chrome** at least:
```
element(by.css('path to select')).sendKeys('value i want');
```

**Comment by JohnCashmore** [comment]
Did anyone find a solution to this for Angular2?

**Comment by JohnCashmore** [comment]
We are also having this issue in Chrome

**Comment by NathanWalker** [comment]
@JohnCashmore this was the only thing I could get to work in Chrome w/ Angular2:
```
element(by.css('path to select')).sendKeys('value i want');
```
I don't use Firefox for integration testing, so not sure there.

**Comment by JohnCashmore** [comment]
Yeah this is what we have done, was hoping there was a solution to using click's in chrome

**Comment by bharathp** [comment]
In firefox, I've notice that the "selected" attribute will not work unless you place the select inside a form, where the form has a name attribute.

It worked for me.

**Comment by shankar14** [comment]
i need dropdown onchange 5 => 5, 10 => 10 , 15 => 15, All => All to display records...row count in show button in yii2
![nshoppy](https://user-images.githubusercontent.com/26651852/27117222-38fc99c2-50f3-11e7-9783-2c6b775d143c.png)

**Comment by shankar14** [comment]
i need dropdown onchange 5 => 5, 10 => 10 , 15 => 15, All => All to display records...row count in show button in yii2

**Comment by vitiral** [comment]
this still doesn't seem to work for the python driver in PhantomJS or Firefox. Any update on this?

**Comment by Saqibm128** [comment]
Not the same as a click, but i use this script to just hack around the issue:

`function(select, option) {
        var script = function(elementLocator, option) {
          option.selected = true;
          option.dispatchEvent(new Event("input"));
          option.dispatchEvent(new Event("change"));
          elementLocator.dispatchEvent(new Event("input"));
          elementLocator.dispatchEvent(new Event("change"));
        }
        return browser.executeScript(script, select, option)`
